EBAY 1.12.2 Manage EAN / UPC / GTIN

thoma202 opened the pull request:
Being abble to manage EAN / UPC/ GTIN 
* Mandatory items specifics do not have a default

Mekki78 commented:
Hello,

This PR require to correct the main errors you have on our validator. All the red one, some of them are really important like hard coded databases or images types.

Please correct this following sections in priority on your module.
![download 12](https://cloud.githubusercontent.com/assets/6247743/12982659/062f9a8c-d0e6-11e5-947a-72e58fad8c1c.png)

Have a great day,

thoma202 commented:
Hi @Mekki78 

I've corrected three errors. Concerning the other errors, you can find the reason why we cannot correct them here : 

![image](https://cloud.githubusercontent.com/assets/2963059/13003552/7efd7804-d176-11e5-996b-d27d3bc038c7.png)

For compatibiliy 1.4 -> 1.6 issues we need to use a class outside of the PrestaShop Scope. This is why we duplicated the Tools class, and added these methods. We need those to 

![image](https://cloud.githubusercontent.com/assets/2963059/13003561/863d9f7c-d176-11e5-8951-b212fc690edb.png)

'ps' here does not correspond to the prefix of the table name but it correspond to the field name, so this is not an error 

![image](https://cloud.githubusercontent.com/assets/2963059/13003565/8e299394-d176-11e5-95f7-1516f9c291f8.png)

For initialization and compatibility 1.4 - 1.6 we use images names, but the user has the choice after to modify it. We need to initilalize the image size used so we have to define them hardcoded. This is a must have in the module. 

![image](https://cloud.githubusercontent.com/assets/2963059/13003571/9a5c2050-d176-11e5-8146-f8f3fe76d232.png)


Once again we are talking about field name and not prefix
